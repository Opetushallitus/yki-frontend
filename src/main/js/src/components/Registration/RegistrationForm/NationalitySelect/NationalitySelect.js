import { Field } from 'formik';
import PropTypes from 'prop-types';
import * as R from 'ramda';
import React from 'react';
import { withTranslation } from 'react-i18next';

import classes from '../RegistrationForm.module.css';

export const nationalitySelect = props => {
  const nationalitiesByLocale = props.nationalities.map(n => {
    const metadata = n.metadata.find(
      m => m.kieli === props.i18n.language.toUpperCase(),
    );
    return { name: metadata.nimi, code: n.koodiArvo };
  });
  const sortByName = R.sortBy(R.prop('name'));
  const nationalityOptions = sortByName(nationalitiesByLocale).map(n => {
    return (
      <option value={n.code} key={n.code}>
        {n.name}
      </option>
    );
  });

  return (
    <div className={classes.InputFieldWrapper}>
      <label htmlFor="select-nationality">
        {props.t('registration.form.nationality')} *
      </label>
      <Field
        id="select-nationality"
        aria-required
        component="select"
        name="nationality"
        className={props.className}
        data-cy="select-nationality"
      >
        <option value="" key="">
          {props.t('common.selectorDefault')}
        </option>
        {nationalityOptions}
      </Field>
    </div>
  );
};

nationalitySelect.propTypes = {
  nationalities: PropTypes.array.isRequired,
  className: PropTypes.string.isRequired,
};

export default withTranslation()(nationalitySelect);
