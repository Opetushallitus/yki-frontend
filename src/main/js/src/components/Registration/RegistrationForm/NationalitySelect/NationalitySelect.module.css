.Select {
    height: 35px;
    width: 250px;
    border: 1px solid hsl(0, 0%, 68%);
    border-radius: 2px;
    background-color: hsl(0, 0%, 100%);
  }