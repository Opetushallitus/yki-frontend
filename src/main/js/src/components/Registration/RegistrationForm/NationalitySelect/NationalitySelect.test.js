import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { nationalitySelect as NationalitySelect } from './NationalitySelect';

configure({ adapter: new Adapter() });

describe('<NationalitySelect />', () => {
  it('should render natonality selection sorted alphabetically', () => {
    const nationalities = [
      {
        koodiArvo: '158',
        metadata: [
          {
            nimi: 'Taiwan',
            kieli: 'FI',
          },
        ],
      },
      {
        koodiArvo: '064',
        metadata: [
          {
            nimi: 'Bhutan',
            kieli: 'FI',
          },
        ],
      },
    ];

    const wrapper = shallow(
      <NationalitySelect
        t={key => key}
        lng={'fi'}
        nationalities={nationalities}
        className="test"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
