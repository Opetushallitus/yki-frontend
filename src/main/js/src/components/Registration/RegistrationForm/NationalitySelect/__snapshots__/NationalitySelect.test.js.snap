// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NationalitySelect /> should render natonality selection sorted alphabetically 1`] = `
<div
  className="InputFieldWrapper"
>
  <label
    htmlFor="select-nationality"
  >
    registration.form.nationality
     *
  </label>
  <FormikConnect(FieldInner)
    aria-required={true}
    className="test"
    component="select"
    data-cy="select-nationality"
    id="select-nationality"
    name="nationality"
  >
    <option
      key=""
      value="placeholder"
    >
      common.selectorDefault
    </option>
    <option
      key="064"
      value="064"
    >
      Bhutan
    </option>
    <option
      key="158"
      value="158"
    >
      Taiwan
    </option>
  </FormikConnect(FieldInner)>
</div>
`;
